PublishParasoftResults/CoverageReportService.ts

```diff
@@ -0,0 +1,279 @@
+/*
+ * Copyright 2024 Parasoft Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as fs from 'fs';
+import * as sax from 'sax';
+import * as lodash from 'lodash';
+import * as tl from 'azure-pipelines-task-lib';
+
+type CoberturaCoverage = {
+    lineRate: number;
+    linesCovered: number;
+    linesValid: number;
+    version: string;
+    packages: Map<string, CoberturaPackage>;
+}
+
+type CoberturaPackage = {
+    name: string;
+    lineRate: number;
+    classes: Map<string, CoberturaClass>;
+}
+
+type CoberturaClass = {
+    fileName: string;
+    name: string;
+    lineRate: number;
+    classId: string;
+    lines: CoberturaLine[];
+}
+
+type CoberturaLine = {
+    lineNumber: number;
+    lineHash: string;
+    hits: number;
+}
+
+export class CoverageReportService {
+    private readonly mergedCoberturaReportPath: string;
+
+    constructor() {
+        this.mergedCoberturaReportPath = tl.getVariable('System.DefaultWorkingDirectory') + '/parasoft-merged-cobertura.xml';
+    }
+
+    mergeCoberturaReports = (reportPaths: string[]): string => {
+        if(reportPaths.length == 1) {
+            fs.copyFileSync(reportPaths[0], this.mergedCoberturaReportPath);
+            return this.mergedCoberturaReportPath;
+        }
+
+        let baseCoverage = this.processXMLToObj(reportPaths[0]);
+        for(let i = 1; i < reportPaths.length; i++) {
+            const reportToMerge: CoberturaCoverage = this.processXMLToObj(reportPaths[i]);
+            try {
+                tl.debug(`Merging cobertura reort: ${reportPaths[i]}`);
+                baseCoverage = this.mergeCoberturaCoverage(lodash.cloneDeep(baseCoverage), reportToMerge);
+            } catch (error) {
```

Comment:
It appears that both Saxon parsing and conflicts during report merging can cause an `Error`. 
Therefore, a generic error message for that is fine. Let's change it to: `Skipped merging Cobertura report "xxx": ${error.message}`
Why do we consider the `else` case as a conflicting case?